examples/shell/cmd_btp.cpp

```diff
@@ -0,0 +1,192 @@
+    }
+    else
+    {
+        ExitNow(error = CHIP_ERROR_INVALID_ARGUMENT);
+    }
+
+exit:
+    return error;
+}
+
+int cmd_btp_scan(int argc, char ** argv)
+{
+    CHIP_ERROR error  = CHIP_NO_ERROR;
+    streamer_t * sout = streamer_get();
+
+    if (argc == 0)
+    {
+        ExitNow(error = CHIP_ERROR_INVALID_ARGUMENT);
+    }
+
+    if (strcmp(argv[0], "start") == 0)
+    {
+        streamer_printf(sout, "Starting scanning over BLE");
+        // start scanning
+        SuccessOrExit(error = CHIP_NO_ERROR);
+    }
+    else if (strcmp(argv[0], "stop") == 0)
+    {
+        streamer_printf(sout, "Stopping scanning over BLE");
+        // stop scanning
+        SuccessOrExit(error = CHIP_NO_ERROR);
+    }
+    else
+    {
+        ExitNow(error = CHIP_ERROR_INVALID_ARGUMENT);
+    }
+
+exit:
+    return error;
+}
+
+int cmd_btp_connect(int argc, char ** argv)
+{
+    CHIP_ERROR error  = CHIP_NO_ERROR;
+    streamer_t * sout = streamer_get();
+
+    if (argc == 0)
+    {
+        ExitNow(error = CHIP_ERROR_INVALID_ARGUMENT);
+    }
+
+    if (strcmp(argv[0], "start") == 0)
+    {
+        streamer_printf(sout, "Connecting to the device over BLE");
+        // connecting
+        SuccessOrExit(error = CHIP_NO_ERROR);
+    }
+    else if (strcmp(argv[0], "stop") == 0)
+    {
+        streamer_printf(sout, "Disconnecting from the device");
+        // disconnecting
+        SuccessOrExit(error = CHIP_NO_ERROR);
+    }
+    else
+    {
+        ExitNow(error = CHIP_ERROR_INVALID_ARGUMENT);
+    }
+
+exit:
+    return error;
+}
+
+int cmd_btp_send(int argc, char ** argv)
+{
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+}
+
+int cmd_btp_dispatch(int argc, char ** argv)
+{
+    CHIP_ERROR error = CHIP_NO_ERROR;
+
+    VerifyOrExit(argc > 0, error = CHIP_ERROR_INVALID_ARGUMENT);
+
+    error = sShellDeviceSubcommands.ExecCommand(argc, argv);
+
+exit:
+    return error;
+}
+
+static const shell_command_t cmds_btp_root = { &cmd_btp_dispatch, "btp", "BLE transport commands" };
+
+/// Subcommands for root command: `btp <subcommand>`
+static const shell_command_t cmds_btp[] = {
+    { &cmd_btp_help, "help", "Usage: btp <subcommand>" },
+    { &cmd_btp_scan, "scan", "Enable or disable scan. Usage: btp scan <start timeout|stop>" },
+    { &cmd_btp_connect, "connect", "Connect or disconnect to a device. Usage: btp connect <start address|stop>" },
+    { &cmd_btp_adv, "adv", "Enable or disable advertisement. Usage: device dump" },
+    { &cmd_btp_send, "send", "Send binary data. Usage: device dump" },
+};
+
+#endif // CONFIG_NETWORK_LAYER_BLE
+
+void cmd_btp_init(void)
+{
+#if CONFIG_NETWORK_LAYER_BLE > 0
+    // CHIP_ERROR error = CHIP_NO_ERROR;
+
+    // Register `device` subcommands with the local shell dispatcher.
+    sShellDeviceSubcommands.RegisterCommands(cmds_btp, ARRAY_SIZE(cmds_btp));
+
+    // Register the root `btp` command with the top-level shell.
+    shell_register(&cmds_btp_root, 1);
+
+    // streamer_t * sout = streamer_get();
```

Comment:
I would prefer to remove commented out code throughout (e.g. in the cpp below as well, where header includes are commented out)

Reply:
Agree